app/export_converters.go

```diff
@@ -158,10 +163,15 @@ func ImportLineForPost(post *model.PostForExport) *LineImportData {
 }
 
 func ImportLineForDirectPost(post *model.DirectPostForExport) *LineImportData {
+	var channelMembers []string = make([]string, len(*post.ChannelMembers))
+	copy(channelMembers, (*post.ChannelMembers))
+	if len(channelMembers) == 1 {
+		channelMembers = append(channelMembers, channelMembers[0])
+	}
```

Comment:
If I'm understanding this correctly, the behaviour is to check if there's only 1 person in the DM channel members list, and if so, to add that person twice.

Is there a reason why this needs to be done this way? It feels a bit weird, but maybe it's necessary?

Reply:
Well, the API [receives two users](https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1direct%2Fpost) in their create DM channel endpoint. Debugging how the self DM channel is created I'm seeing that we're sending the same user twice and the channel is created with `userid_userid` name.

I've tried to follow our API behavior, for consistency, having the same user twice in the export JSON so we have it in the same way as our API.

Also I wanted to make a copy of the slice in order to avoid side-effects